gr-utils/modtool/support/_gr_modtool

```diff
@@ -0,0 +1,25 @@
+#compdef gr_modtool
+#autoload
+
+# SPDX-License-Identifier: LGPL-3.0-or-later
+# Copyright 2021 Marcus Müller
+# This module offers tab-completion for ZSH
+# It's rudimentary.
+
+local -a subcmds
+subcmds=(
+    'add:Adds a block to the out-of-tree module.'
+    'bind:Generate Python bindings for GR block'
+    'disable:Disable selected block in module.'
+    'info:Return information about a given module'
+    'makeyaml:Generate YAML files for GRC block bindings.'
+    'newmod:Create new empty module, use add to add blocks.'
+    'rename:Rename a block inside a module.'
+    'rm:Remove a block from a module.'
+    'update:Update the grc bindings for a block'
+)
+_describe 'command' subcmds
+if (( CURRENT == 2 )); then
+    _describe -t commands 'subcommands' subcmds
+fi
+return 0
```

Comment:
Don't unconditionally return zero. That will be taken to mean that the function has done something and there is no need to carry on and try something else like retrying with fuzzy matching, approximations, case-insensitivity or whatever the user has configured.

Usually you can just rely on the fact that a shell function passes on the return status from whatever the last command was, e.g. `_describe`. If the last thing was an if then else, or case statement, that still works correctly, passing on the status from the last command in whichever branch was taken.

But if you look at examples, it is common to accumulate a return status in a `$ret` variable.